prover/src/provers/mod.rs

```diff
@@ -17,3 +17,47 @@ use anyhow::Result;
 pub trait Prover<T> {
     fn prove(&self, ctx: &T) -> Result<()>;
 }
+
+const DEGREE_BITS_RANGE: [[std::ops::Range<usize>; 6]; 5] = [
+    [10..21, 10..15, 10..18, 8..15, 10..21, 15..23],
+    [10..21, 12..22, 13..21, 8..21, 10..21, 13..23],
+    [10..21, 12..22, 13..21, 8..21, 10..21, 13..23],
+    [10..21, 12..22, 13..21, 8..21, 10..21, 13..23],
+    [10..21, 12..22, 13..21, 8..21, 10..21, 13..23],
+];
+// const DEGREE_BITS_RANGE: [[std::ops::Range<usize>; 6]; 5] = [
+//     [16..17, 12..13, 10..16, 9..12, 15..17, 17..19],
+//     [16..17, 15..17, 12..19, 9..14, 15..17, 19..20],
+//     [16..17, 15..17, 12..19, 11..14, 16..19, 19..21],
+//     [16..17, 17..18, 14..19, 11..14, 16..19, 21..22],
+//     [16..18, 16..20, 16..21, 14..15, 18..21, 21..23],
+// ];
+
+fn select_degree_bits(seg_size: usize) -> [std::ops::Range<usize>; 6] {
+    let seg_size_to_bits = std::collections::BTreeMap::from([
+        (1024, 0),
```

Comment:
We can define this HashMap in the mod namespace, and when you check the seg_size, you can get the keys of the map.Hence, we don't have to hardcode the bits in many places.